frontend/src/Components/EventCard.jsx

```diff
@@ -4,19 +4,22 @@ import axios from 'axios';
 import moment from 'moment';
 
 
-const EventsCard = (props) => {
+const EventCard = (props) => {
     const history = useHistory();
     const { setFeedback, loggedUser, event } = props;
 
     const [ volunteerHours, setVolunteerHours ] = useState('');
-    const [ volunteersList, setVolunteersList] = useState([]);
+    const [ volunteersList, setVolunteersList ] = useState([]);
+    const [ waitingForRender, setWaitingForRender ] = useState(false);
     
     useEffect(() => {
         setVolunteersList(event.volunteersList);
+        setWaitingForRender(false)
     }, [event.reload, event.volunteersList]);
 
     const manageVolunteersRequests = async (e, volunteerId) => {
         try {
+            setWaitingForRender(true);
             await axios.patch(`/api/event_attendees/event/${event.event_id}/volunteer/${volunteerId}`, {confirmed: e.target.checked});
             event.setReload(event.reload + 1);
```

Comment:
not something you've changed this pr but this should be !event.reload like the others when the original is a bool

Reply:
not really, imagine this is a spinner, at the beginning it's false, when you're about to do a network request, it's set to true, the network request is triggered then the sibling component will render through `event.setReload` changing the event which on its turn will trigger `useEffect(() => {
        setVolunteersList(event.volunteersList);
        setWaitingForRender(false)
    }, [event.reload, event.volunteersList]);` 
setting it back to false (kind of completing the cycle)

Reply:
oh, not line 22. i meant line 24, event.setReload(event.reload + 1);